graphqlapi/src/main/java/io/stargate/graphql/graphqlservlet/HTTPAwareContextImpl.java

```diff
@@ -34,6 +40,14 @@
   private HttpServletRequest request;
   private HttpServletResponse response;
 
+  // We need to manually maintain state between multiple selections in a single mutation
+  // operation to execute them as a batch.
+  // Currently graphql-java batching support is only restricted to queries and not mutations
+  // See https://www.graphql-java.com/documentation/v15/batching/ and
+  // https://github.com/graphql-java/graphql-java/blob/v15.0/src/main/java/graphql/execution/instrumentation/dataloader/DataLoaderDispatcherInstrumentation.java#L112-L115
+  // For more information.
+  private final BatchContext batchContext = new BatchContext();
```

Comment:
This is a pretty complex object (with it allocating many objects itself) to be allocating for non-batch related operations. I wonder if there's a good way to only incur the expense for batch operations.

Reply:
We could use get-or-create semantics but I don't know if its worth the complexity to prevent additional allocations in GraphQL. GraphQL library is already creating a huge context per request...